.github/workflows/ci.yml

```diff
@@ -101,83 +115,51 @@ jobs:
         # see https://github.com/conda-incubator/setup-miniconda/#important
         shell: bash -l {0}
 
-    env:
-      IS_MASTER_BUILD: ${{ !matrix.numfocus_nightly && matrix.os == 'ubuntu-latest' && matrix.pyarrow == '4.0.1' && matrix.python == '3.8' }}
-      IS_TAG: ${{ github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')}}
-
     steps:
-      # CI setup
-      - name: Is Master Build?
-        run: echo $IS_MASTER_BUILD
-      - name: Checkout
-        uses: actions/checkout@v4
-      - name: Set up Conda env
-        uses: mamba-org/setup-micromamba@068f1ab4b37ed9b3d9f73da7db90a0cda0a48d29
+      - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
+      - name: Set up pixi
+        uses: prefix-dev/setup-pixi@ba3bb36eb2066252b2363392b7739741bb777659 # v0.8.1
         with:
-          environment-file: environment.yml
-          cache-environment: true
-          create-args: >-
-            python=${{ matrix.python }}
+          environments: ${{ matrix.python }} # References the environments defined in pixi.toml
       - name: Install Pyarrow (non-nightly)
-        run: micromamba install -y pyarrow==${{ matrix.pyarrow }}
+        run: pixi add pyarrow=${{ matrix.pyarrow }}
         if: matrix.pyarrow != 'nightly' && matrix.pandas == ''
       - name: Install Pyarrow (nightly)
         # Install both arrow-cpp and pyarrow to make sure that we have the
         # latest nightly of both packages. It is sadly not guaranteed that the
         # nightlies and the latest release would otherwise work together.
-        run: micromamba update -c arrow-nightlies -c conda-forge arrow-cpp pyarrow
+        # FIXME: The nightlies channel is currently empty: https://github.com/apache/arrow/issues/41856
```

Comment:
I was a bit surprised that adding the channel arrow-nightlies is (atm) completely useless.

Reply:
As long as the conda channel is empty, use pip:
https://arrow.apache.org/docs/developers/python.html#installing-nightly-packages:~:text=pip%20install%20%2D%2Dextra%2Dindex%2Durl%20https%3A//pypi.fury.io/arrow%2Dnightlies/%20%5C%0A%20%20%20%20%2D%2Dprefer%2Dbinary%20%2D%2Dpre%20pyarrow

(?)